lib/CGameState.cpp

```diff
@@ -3300,9 +3300,6 @@ void CPathfinder::initializeGraph()
 
 void CPathfinder::calculatePaths()
 {
-	assert(hero);
-	assert(hero == getHero(hero->id));
-
 	bool flying = hero->hasBonusOfType(Bonus::FLYING_MOVEMENT);
```

Comment:
As it's not really related to this pull request I wouldn't touch it for now.
Once I get my hands over flying and water walking I'll check it.